lib/src/mock_query.dart

```diff
@@ -122,6 +128,37 @@ class MockQuery extends Mock implements Query {
     });
   }
 
+  @override
+  Query startAt(List<dynamic> values) =>
+      _cursorUtil(values, (docs, index) => docs.sublist(max(0, index)));
+
+  @override
+  Query endAt(List<dynamic> values) => _cursorUtil(values,
+      (docs, index) => docs.sublist(0, index == -1 ? docs.length : index));
+
+  Query _cursorUtil(
+    List<dynamic> values,
+    List<DocumentSnapshot> Function(List<DocumentSnapshot> docs, int index) f,
+  ) {
+    return MockQuery(this, (docs) {
+      assert(
+        parameters['orderedBy'] != null &&
+            parameters['orderedBy'].length >= values.length,
+        'You can only specify as many start values as there are orderBy filters.',
+      );
+      if (docs.isEmpty) return docs;
+
+      var res = List.of(docs);
+      for (var i = 0; i < values.length; i++) {
+        final index = docs.indexWhere(
+          (doc) => doc.data()?[parameters['orderedBy'][i]] == values[i],
+        );
+        res = f(docs, index);
```

Comment:
Should it be `f(docs,...` or `f(res,...`?

Reply:
Yea, I think it's supposed to be f(res,...) actually.